### PR TITLE
FakeNitro: Add app icon customization

### LINKED ISSUE #1811: [Feature Request] Update FakeNitro, with the custom icons feature
## Discord Account
sallyqwp

## What is it that you'd like to see?
Just update FakeNitro with the custom in-app icons

### Request Agreement

- [X] I have read the requirements for opening an issue above

### PR DESCRIPTION
**Pull request by zImPatrick**
fixes #1811

i suck at regex

### CODE REVIEW

File: src/plugins/fakeNitro/index.ts [hunk 1]
```diff
@@ -329,6 +335,22 @@ export default definePlugin({
...
+            replacement: {
+                match: /canUsePremiumAppIcons:function\(\i\){/,
+                replace: "$&return true;"
+            }
+        },
+        {
+            find: "location:\"AppIconHome\"",
+            predicate: () => settings.store.enableAppIcons,
+            replacement: {
+                match: /(,.=).\[2\]/,
```
Comment (ImLvna) [inline src/plugins/fakeNitro/index.ts:351]: this match looks extremely loose

**Review by ImLvna — APPROVED** [review]

File: src/plugins/fakeNitro/index.ts (hunk 1 above)
Comment (zImPatrick) [reply src/plugins/fakeNitro/index.ts:351]: what would be a better regex to match `, l = o[2]`?

alternatively I could also target `Lr.ZP.isPremium` with something like `.{1,2}\..{1,2}\.isPremium` I think, would that be better?

Comment (ImLvna) [reply src/plugins/fakeNitro/index.ts:351]: PLEASE dont use `.` much in regexes. if you need to match a variable name, use `\i`. Please read the plugin guide first: https://github.com/Vendicated/Vencord/blob/main/docs/2_PLUGINS.md

Comment (TheKodeToad) [reply src/plugins/fakeNitro/index.ts:351]: +1

You can also use lookahead and lookbehind to match some of the context

File: src/plugins/fakeNitro/index.ts [hunk 2]
```diff
@@ -159,6 +159,12 @@ const settings = definePluginSettings({
         type: OptionType.BOOLEAN,
         default: true,
         restartNeeded: true
+    },
+    enableAppIcons: {
```
Comment (Vendicated) [inline src/plugins/fakeNitro/index.ts:163]: imo having a setting for this is pointless

**Comment by Vendicated** [comment]
thanks!

**Comment by Sallyqwp** [comment]
LET'S GO

**Pull request merged**